inc/css/blocks/class-core-image-plugin-enhancement.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Css handling logic for blocks.
+ *
+ * @package ThemeIsle\GutenbergBlocks\CSS\Blocks
+ */
+
+namespace ThemeIsle\GutenbergBlocks\CSS\Blocks;
+
+use ThemeIsle\GutenbergBlocks\Base_CSS;
+
+use ThemeIsle\GutenbergBlocks\CSS\CSS_Utility;
+
+use Masterminds\HTML5;
+
+/**
+ * Class Core_Image_Plugin_CSS
+ */
+class Core_Image_Plugin_CSS extends Base_CSS {
+
+	/**
+	 * The library under which the blocks are registered.
+	 *
+	 * @var string
+	 */
+	public $library_prefix = 'core';
+
+	/**
+	 * The namespace under which the blocks are registered.
+	 *
+	 * @var string
+	 */
+	public $block_prefix = 'image';
+
+	/**
+	 * Generate Icon Block CSS
+	 *
+	 * @param mixed $block Block data.
+	 * @return string
+	 * @since   1.7.0
+	 * @access  public
+	 */
+	public function render_css( $block ) {
+		$css = new CSS_Utility( $block );
+
+		$html5  = new HTML5();
+		$dom    = $html5->loadHTML( $block['innerHTML'] );
```

Comment:
@Soare-Robert-Daniel this looks like an expensive operation for fetching the ID of the element. 

Can't we pass it using the block props or similar?

Reply:
This is @HardeepAsrani implementation, it looks more cleaner than my previous one: https://github.com/Codeinwp/gutenberg-blocks/pull/772/commits/dbd065e4e5dea4681c6b75676248c6abcd697b7f#diff-5dd7c5cd584c4495a9422901ec25d408c4f384906506cb226ca925fadaf6c9a4R835-R844

I had no idea about the performance.

Reply:
It tried and that part takes around 3-4 milliseconds. What do you think? @selul

Reply:
again, it seems too much for me just to fetch the id, which can be transferred as block meta prop or any alternative.

Reply:
makes sense, we can review later on then.

Reply:
I might be able to do something, let me check and let you know in few minutes.

Reply:
A simple alternative approach will be this:

```php
$id_pos =  strpos( $block['innerHTML'], "wp-block-themeisle-blocks-image-" );

if ( false === $id_pos ) {
	return $style;
}

$id = substr( $block['innerHTML'], $id_pos, strlen( "wp-block-themeisle-blocks-image-" ) + 8;
```

*But a big disadvantage is that the formula is hardcoded, which creates a compatibility issue in the future if we change the ID format.*

A second alternative will be with Regex: https://regex101.com/r/BPUu0v/2

```php
$re = '/<figure.*id="([\w-]*)">/m';
preg_match_all($re, $block['innerHTML'], $matches, PREG_SET_ORDER, 0);

if ( ! isset($matches[0]) || ! isset($matches[0][1]) ) {
	return $style;
}

$id = $matches[0][1];
```

@selul, what do you think?

Reply:
Lol, I was also making a regex for it. Thank you for saving me from the horror. XD

Reply:
we can't send it or attach the id from the frontend when we edit the block, as a meta prop?

Reply:
@selul We can but we will have to add an extra attribute to it. By default they have an ID attribute which is an HTML selector so we can't get it in PHP, it's just part of the HTML. But regex should work just fine, even if tomorrow the user decides to change the ID to a custom one.

Reply:
Currently, we take ID from the _anchor_ attribute.

Reply:
@selul, we use the built-in support for [anchor](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/image/block.json#L79). The disadvantage of this is the anchor is not sent as a separate attribute, that we need to get the values using other methods. 

@HardeepAsranim @selul should we stick with Regex?

Reply:
@Soare-Robert-Daniel regex is good, but not perfect and sometimes it failed to parse certain cases, see https://regex101.com/r/b7FVda/1/ where I've added different variations of id attribute in figure. As you can see only a very specific case it successfully parse. 

I would say if regex is the only option, to drop it in favor of the current implementation.

Reply:
Something like this will work better ```/<figure.*?id="([^"]*?)".*?>/mg``` https://regex101.com/r/5HmvSO/1

WordPress only uses double quotes so we can ignore single quote case.

Reply:
makes sense, we can ignore the `g` flag when using since it will have only one occurrence I guess?

Reply:
Yea, that's only for testing on regex101.

@Soare-Robert-Daniel can you take care of it and ask Irinel for a review?

Reply:
> Something like this will work better `/<figure.*?id="([^"]*?)".*?>/mg` https://regex101.com/r/5HmvSO/1
> 
> WordPress only uses double quotes so we can ignore single quote case.

The `?` does not seems to be necessary.

Reply:
With `?` it does a lazy match, with just a period it does a greedy match. In principle, it should be faster with a question mark. Obviously, our matches aren't as complex but yea, that's a question mark. Shouldn't make much difference in our case.